catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/util/AttributeFactory.java

```diff
@@ -0,0 +1,193 @@
+        return doCreate(attributeDescriptor, Collections.singletonList(attributeValue));
+    }
+
+    /**
+     * Attempts to parse a value for an {@link Attribute} according to the provided {@link AttributeDescriptor}
+     * whose value is represented by the given string {@param value}. Returns {@code null} if {@param value}
+     * could not be parsed.
+     * <p>
+     * The input {@param value} must be effectively {@link Serializable}.
+     *
+     * @param attributeDescriptor The descriptor to use to parse the value.
+     * @param value               The string containing the value to be parsed.
+     * @return The deserialized object of {@param value} according to {@param attributeDescriptor}.
+     */
+    @Nullable
+    public Serializable parseAttributeValue(AttributeDescriptor attributeDescriptor, String value) {
+        try {
+            notNull(attributeDescriptor);
+            notEmpty(value);
+
+            Serializable deserializedValue;
+            AttributeType attributeType = attributeDescriptor.getType();
+            AttributeType.AttributeFormat attributeFormat = attributeType.getAttributeFormat();
+
+            switch (attributeFormat) {
+
+            case INTEGER:
+                deserializedValue = Integer.parseInt(value);
+                break;
+
+            case FLOAT:
+                deserializedValue = Float.parseFloat(value);
+                break;
+
+            case DOUBLE:
+                deserializedValue = Double.parseDouble(value);
+                break;
+
+            case SHORT:
+                deserializedValue = Short.parseShort(value);
+                break;
+
+            case LONG:
+                deserializedValue = Long.parseLong(value);
+                break;
+
+            case DATE:
+                Calendar calendar = DatatypeConverter.parseDateTime(value);
+                deserializedValue = calendar.getTime();
+                break;
+
+            case BOOLEAN:
+                deserializedValue = Boolean.parseBoolean(value);
+                break;
+
+            case BINARY:
+                deserializedValue = value.getBytes(Charset.forName("UTF-8"));
+                break;
+
+            case OBJECT:
+            case STRING:
+            case GEOMETRY:
+            case XML:
+                deserializedValue = value;
+                break;
+
+            default:
+                return null;
+            }
+
+            return deserializedValue;
+
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Any time a single string is presented for any {@link AttributeDescriptor} where that descriptor's
+     * {@link AttributeDescriptor#isMultiValued()} is true, this method defines the parsing strategy.
+     * <p>
+     * The default is to treat the string as a comma-separated list of values.
+     * <p>
+     * <i>This method could get promoted to <b>protected</b> in the future.</i>
```

Comment:
Imho, I would either make it protected right now or, to be on the safe side, keep it private and not mention anything. If someone down the road needs to allow extension of this class, then they'll be in better position to expose what's really needed at that time. The comment doesn't hurt anything really, but I don't think it helps either and may become confusing down the road as the code evolves, especially if someone forgets to update the comment.